warehouse/backend_config.go

```diff
@@ -20,32 +22,32 @@ import (
 
 // TODO: add tests
 func newBackendConfigManager(
-	c *config.Config, // TODO possibly use this to get all the needed variables
-	wrappedDB *sqlquerywrapper.DB,
+	c *config.Config,
+	db *sqlquerywrapper.DB,
 	bc backendconfig.BackendConfig,
-	l logger.Logger,
+	log logger.Logger,
 ) *backendConfigManager {
 	if c == nil {
 		c = config.Default
 	}
 	if bc == nil {
 		bc = backendconfig.DefaultBackendConfig
 	}
-	if l == nil {
-		l = logger.NOP
+	if log == nil {
+		log = logger.NOP
 	}
 	bcm := &backendConfigManager{
 		conf:                 c,
-		db:                   wrappedDB,
-		schema:               repo.NewWHSchemas(wrappedDB),
+		db:                   db,
+		schema:               repo.NewWHSchemas(db),
 		backendConfig:        bc,
-		logger:               l,
+		logger:               log,
 		initialConfigFetched: make(chan struct{}),
 		connectionsMap:       make(map[string]map[string]model.Warehouse),
 	}
-	if config.GetBool("ENABLE_TUNNELLING", true) {
+	if c.GetBool("ENABLE_TUNNELLING", true) {
 		bcm.internalControlPlaneClient = cpclient.NewInternalClientWithCache(
-			backendconfig.GetConfigBackendURL(),
+			c.GetString("CONFIG_BACKEND_URL", "https://api.rudderstack.com"),
```

Comment:
Why do we keep duplicating the variable name and its default? Isn't `GetConfigBackendURL` good enough?

Reply:
I wanted to avoid calling global function which for backendconfig was difficult to test.

Reply:
Right, because you replaced the bc with the tenant manager. Let's keep it this way for now then :+1: